src/core/unit.c

```diff
@@ -1868,6 +1868,10 @@ int unit_start(Unit *u) {
 
         assert(u);
 
+        /* Let's hold off running start jobs for mount units when /proc/self/mountinfo monitor is rate limited. */
+        if (u->type == UNIT_MOUNT && sd_event_source_is_ratelimited(u->manager->mount_event_source))
+                return -EAGAIN;
```

Comment:
Maybe, it is better to introduce a function in unit vtable, e.g. `.can_start_early()`??

Reply:
I don't know if it is worth adding virtual function if we now that only mount unit type would provide an implementation.